### PR TITLE
MSEARCH-514 Add new field to index to support CN browse by types

### PR DESCRIPTION
**Pull request by psmagin**
## Purpose
Add a new field to the instance index to support call-numbers browse by types

## Approach
- Add typedCallNumber to index mapping
- Extend item schema to have callNumberTypeId
- Add processor to the new field in index, use reference service to load `local` call-number types, and implement new call-number to long conversion method to have reserved 1st digit for CN type
- Add new permission for system-user

### Changes checklist
- [ ] API paths, methods, request or response bodies changed, added, or removed
- [ ] Database schema changes
- [ ] Interface versions changes
- [ ] Interface dependencies added, or removed
- [ ] Permissions changed, added, or removed
- [ ] Check logging

### CODE REVIEW

**Review by Shans-Kaluhin — APPROVED** [review]

**Review by hasanepam — APPROVED** [review]

File: src/main/java/org/folio/search/utils/CallNumberUtils.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    long startVal = convertChar(firstPosition, CN_MAX_CHARS);
+    return callNumberToLong(callNumber, startVal, CN_MAX_CHARS - 1);
+  }
+
+  private static long callNumberToLong(String callNumber, long startVal, int maxChars) {
+    var cleanCallNumber = cleanCallNumber(callNumber, maxChars);
+    if (StringUtils.isBlank(cleanCallNumber)) {
+      return 0L;
+    }
+    long result = startVal;
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/utils/CallNumberUtils.java:149]: image

File: src/main/java/org/folio/search/utils/CallNumberUtils.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+import java.util.stream.Stream;
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.RegExUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.util.Assert;
+
+@UtilityClass
+public class CallNumberUtils {
+
+  private static final int CN_MAX_CHARS = 10;
```
Comment (viacheslavkol) [inline src/main/java/org/folio/search/utils/CallNumberUtils.java:21]: image

**Review by viacheslavkol — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/search/utils/CallNumberUtils.java (hunk 1 above)
Comment (psmagin) [reply src/main/java/org/folio/search/utils/CallNumberUtils.java:149]: Check line 140 in the file, it does what described

File: src/main/java/org/folio/search/utils/CallNumberUtils.java (hunk 2 above)
Comment (psmagin) [reply src/main/java/org/folio/search/utils/CallNumberUtils.java:21]: In method on line 139 there is 'CN_MAX_CHARS-1'

**Review by viacheslavkol — APPROVED** [review]

**Pull request merged**